From 1249267f7aeb4cbf3ba54e4819a08851ad19ae37 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 10 Jun 2019 14:59:42 +0300
Subject: [PATCH] [SW-0] gaudi: pll: fix garbled serial console

Once we add the PCI PLL to the W/A group and switch the DIV2 output,
the serial port output might get garbled because the baud rate might not
be appropriate for the new clock frequency.
Fix this by updating the baud rate after the PCI PLL changes.

Change-Id: Ia259138249c96d9b4e6dff62ed6629904cffcb79
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index ab00579..1735fc1 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -410,5 +410,9 @@
 		}
 
 		hl_pll_wa_clk_sw(&gaudi_pll_cfg[i]);
+
+		/* if we've just switched the UART clock, reinitialize it */
+		if (gaudi_pll_cfg[i].base == mmPSOC_PCI_PLL_BASE)
+			hl_uart_init();
 	}
 }
